### PR TITLE
Feature/notify svn jobs

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Add SVN global hook handler which triggers builds for svn repository commits.

### Your checklist for this pull request

- [ ] PR is well described
- [ ] Related issues linked to PR if existing and labels set
- [ ] Target branch is not master (in most cases develop should bet the target of choice) 
- [ ] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [ ] CHANGELOG.md updated
- [ ] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Review by pfeuffer — CHANGES_REQUESTED** [review]
Just small remarks...

File: src/main/java/sonia/scm/jenkins/CsrfCrumbRequester.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      content = response.contentAsStream();
+      csrfCrumb = CsrfCrumbParser.parse(content);
+    } finally {
+      IOUtil.close(content);
+    }
+    return csrfCrumb;
+  }
+
+  private static String createCrumbUrl(String url) {
+    return escape(createBaseUrl(url).concat("crumbIssuer/api/xml"));
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/java/sonia/scm/jenkins/CsrfCrumbRequester.java:80]: Just copied this part. Not sure if this is really necessary.

File: src/main/java/sonia/scm/jenkins/GlobalJenkinsConfiguration.java [hunk 2]
```diff
@@ -50,9 +52,17 @@
...
   private boolean disableEventTrigger = false;
 
+  private String username;
+
+  @XmlJavaTypeAdapter(XmlEncryptionAdapter.class)
+  private String apiToken;
+
   private String url;
 
   public GlobalJenkinsConfiguration(String url, boolean disableGitTrigger, boolean disableMercurialTrigger, boolean disableRepositoryConfiguration) {
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/java/sonia/scm/jenkins/GlobalJenkinsConfiguration.java:68]: Missed this one. I don't think that we still need it.

File: src/main/java/sonia/scm/jenkins/JenkinsGlobalHookHandler.java [hunk 3]
```diff
@@ -24,210 +24,113 @@
...
-  /** Field description */
-  private GlobalJenkinsConfiguration configuration;
+  private void logGlobalJenkinsHook(String type) {
+    //J-
+    logger.debug(
+      "check for global jenkins hook: type={}, hg disabled={}, git disabled={}",
+      type,
+      configuration.isDisableMercurialTrigger(),
+      configuration.isDisableGitTrigger()
+    );
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/java/sonia/scm/jenkins/JenkinsGlobalHookHandler.java:101]: You are right

File: src/main/js/GlobalJenkinsConfigurationForm.tsx [hunk 4]
```diff
@@ -112,6 +123,23 @@ class GlobalJenkinsConfigurationForm extends React.Component<Props, State> {
...
+        />
+        <InputField
+          name={"apiToken"}
+          type="password"
+          label={t("scm-jenkins-plugin.global.form.apiToken")}
+          helpText={t("scm-jenkins-plugin.global.form.apiTokenHelp")}
+          disabled={readOnly}
+          value={this.state.apiToken}
+          onChange={this.valueChangeHandler}
+        />
```
[the remark this answers is not in the document]
Comment (eheimbuch) [reply src/main/js/GlobalJenkinsConfigurationForm.tsx:142]: I agree

**Pull request merged**